[docs] Update `Transitions` page (#27319)
mui-org/material-ui: docs/src/pages/components/transitions/transitions.md

}
```

> > 💡 Transition components require first child component to forward ref to DOM node. If it cannot find DOM node, it throws error.

Reviewer: The usage of `> ` is not consistent with the first section.

```suggestion
The transition components require the first child element to forward its ref to the DOM node.
```

{{"demo": "pages/components/transitions/SimpleZoom.js", "bg": true}}

## Child requirement

- **Forward the style** : To better support server rendering, Material-UI provides a `style` prop to the children of some transition components, (Fade, Grow, Zoom, Slide). The `style` prop must be applied to the DOM for the animation to work as expected.
- **Forward the ref** : The transition components require the first child element to forward its ref to the DOM node. For more details about ref, check out [Caveat with refs](/guides/composition/#caveat-with-refs)
- **Single element** : The transition components require only 1 child element (`React.Fragment` is not allowed).

```jsx
// The `props` object contains a `style` prop.
// You need to provide it to the `div` element as shown here.
const MyComponent = React.forwardRef(({ style, ...props }, ref) {
  return (

>     <div ref={ref} {...props} style={style}>

Reviewer 2: We already have a comment about forwarfing the `style` prop. I don’t think we need to emphasize again, especially because the `style` prop is not handled in the component.
